app/lib/route_validators/valid_register_api_route.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module RouteValidators
+  class ValidRegisterApiRoute
```

Comment:
Are these classes validators? I suggest we stick with the "constraint" terminology, such that the namespace is `RouteConstraints` and the classes are `SystemAdminConstraint` and `ApiVersionConstraint`